src/stepFunctions/commands/visualizeStateMachine.ts

```diff
@@ -29,257 +29,341 @@ export interface messageObject {
-            command: 'update',
-            stateMachineData: updatedTextDocument.getText(),
-            isValid,
-        })
+class AslVisualization {
+    public readonly documentURI: vscode.Uri
+    public readonly webviewPanel: vscode.WebviewPanel
+    public onVisualizationDisposeEmitter = new vscode.EventEmitter<void>()
+    public readonly onVisualizationDispose = this.onVisualizationDisposeEmitter.event
+    private readonly disposables: vscode.Disposable[]
+    private isPanelDisposed: boolean
+
+    public constructor(textDocument: vscode.TextDocument) {
+        this.documentURI = textDocument.uri
+        this.isPanelDisposed = false
+        this.disposables = []
+        this.webviewPanel = this.setupWebviewPanel(textDocument)
     }
 
-    const debouncedUpdate = debounce(sendUpdateMessage, 500)
-
-    // Set the initial html for the webpage
-    panel.webview.html = getWebviewContent(
-        ext.visualizationResourcePaths.webviewBodyScript.with({ scheme: 'vscode-resource' }),
-        ext.visualizationResourcePaths.visualizationLibraryScript.with({ scheme: 'vscode-resource' }),
-        ext.visualizationResourcePaths.visualizationLibraryCSS.with({ scheme: 'vscode-resource' }),
-        ext.visualizationResourcePaths.stateMachineCustomThemeCSS.with({ scheme: 'vscode-resource' }),
-        {
-            inSync: localize(
-                'AWS.stepFunctions.graph.status.inSync',
-                'Previewing ASL document. <a href="" style="text-decoration:none;">View</a>'
-            ),
-            notInSync: localize('AWS.stepFunctions.graph.status.notInSync', 'Errors detected. Cannot preview.'),
-            syncing: localize('AWS.stepFunctions.graph.status.syncing', 'Rendering ASL graph...'),
+    public getPanel(): vscode.WebviewPanel | void {
+        if (this.webviewPanel && !this.isPanelDisposed) {
+            return this.webviewPanel
         }
-    )
+    }
 
-    // Add listener function to update the graph on document save
-    const updateOnSaveDisposable = vscode.workspace.onDidSaveTextDocument(async savedTextDocument => {
-        if (savedTextDocument && savedTextDocument.uri.path === documentUri.path) {
-            await sendUpdateMessage(savedTextDocument)
+    public getWebview(): vscode.Webview | void {
+        if (this.webviewPanel && this.webviewPanel.webview && !this.isPanelDisposed) {
+            return this.webviewPanel.webview
         }
-    })
+    }
 
-    const onDidCloseTextDocumentDisposable = vscode.workspace.onDidCloseTextDocument(documentWillSaveEvent => {
-        if (!trackedDocumentDoesExist(documentUri)) {
-            panel.dispose()
-            vscode.window.showInformationMessage(
-                localize(
-                    'AWS.stepfunctions.visualisation.errors.rename',
-                    'State machine visualization closed due to file renaming or closure.'
-                )
-            )
-        }
-    })
+    public getOnVisualizationDisposeEvent(): vscode.Event<void> {
```

Comment:
This function is unnecessary with the `onVisualizationDispose` property already public